autoload/lsp/diag.vim

```diff
@@ -210,10 +210,16 @@ def ShowDiagInPopup(diag: dict<any>)
     # The line number is outside the last line in the file.
     dlnum = lastline
   endif
+  # Make sure column is not outside the character in line.
   if dlcol < 1
-    # The column is outside the last character in line.
-    dlcol = ltext->len() + 1
+    dlcol = 1
+  elseif dlcol > ltext->len()
+    dlcol = ltext->len()
   endif
+
+  # Move cursor to the right col of current diag
+  cursor(dlnum, dlcol)
```

Comment:
I don't think this function should move the cursor, it would make sense to change `LspDiagsJump` instead to move the cursor to the correct column, something like https://github.com/yegappan/lsp/pull/165

Reply:
Or possible the cursor movement should be moved to the function `ShowCurrentDiag`.

Reply:
Yes.  The cursor should not be moved.  The popup window with the diag message should be displayed in the correct column.

Reply:
what do you mean? i cannot get your point.
this is to get the diag message to be displayed in the correct column, and more than that, to make the code action took action for the correct one! some lsp server may have code action advise even no diag, unless cursor on correct place.